src/pro_eco_capture_family.rs

```
@@ -0,0 +1,73 @@
+use super::{sys, AudioCaptureFrame, Result, UniversalCaptureFamily};
+use snafu::prelude::*;
+use std::{os::raw::c_longlong, time::Duration};
+
+pub trait ProEcoCaptureFamily: UniversalCaptureFamily {
+    fn event(&self) -> sys::MWCAP_PTR;
+
+    fn get_device_time(&self) -> Result<Duration> {
+        let mut time: c_longlong = 0;
+        unsafe {
+            if sys::MWGetDeviceTime(self.handle(), &mut time as *mut _)
+                != sys::_MW_RESULT__MW_SUCCEEDED
+            {
+                whatever!("unable to get device time");
+            }
+            Ok(Duration::from_nanos(100 * time as u64))
+        }
+    }
+
+    /// Causes `wait` to return any time the specified events (e.g.
+    /// `MWCAP_NOTIFY_AUDIO_FRAME_BUFFERED`) occur. Returns a handle that can be used to
+    /// unregister.
+    fn register_notify(&self, events: u32) -> Result<sys::MWCAP_PTR> {
```

1. As a caller from the crate docs alone I wouldn't know what to pass in for `events`. Maybe wrap with a Rust enum or `bitflags!` (is this a bitfield?) rather than the raw thing? 

2. `MWCAP_PTR` has an `unsafe` smell. If I make up a nonsense value and pass it to `unregister_notify`, do I get Undefined Behavior? I suspect yes, and safe APIs are supposed to be sound even if the caller does the worst thing imaginable, so I think I'd wrap it so the caller can't do that. Similarly, if I pass the handle to the "wrong" channel, do I get Undefined Behavior? That seems trickier to address, but a couple ideas:
    * we could pair the channel handles with our own sequence number, copy that into the notify wrapper, and panic if the two don't match.
    * we could restructure to have the channel handle own the `MWCAP_PTR`; the caller unregisters with either by providing the same `events` back (if it's not a bitfield having exactly one registration per `events` might make sense) or a matching integer handle. [**edit:** I guess really this comes down to keeping a list of the register `MWCAP_PTR`s; we can pass the same bytes to the caller, just check that what they pass back is in the list at unregister time and remove it.]
    * or if you don't want to deal with the complexity of those, just mark `unregister_notify` as `unsafe`?

---

1. Yep, using `bitflags`.
2. I tested with arbitrary values and it seems to work just fine (as in, unregister returns an error). How about I just wrap it in a new type so you can't construct arbitrary values and it doesn't smell so unsafe?

---

+1 to wrap in newtype then to just make it less scary-looking from the crate docs.

---

Done.

---

Oh, I guess my approve was slightly premature because this part isn't done yet

---

I think it is? Make sure you're not looking at the outdated file.